File: CS/E2502/MyDockManager/MyDockElementsSkinPainter.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraBars.Docking.Paint;
using DevExpress.XtraBars.Styles;
using DevExpress.XtraBars.Docking;

namespace WindowsFormsApplication1
{
    public class MyDockElementsSkinPainter : DockElementsSkinPainter
    {
        public MyDockElementsSkinPainter(SkinBarManagerPaintStyle paintStyle) : base(paintStyle) { }
        protected override void CreateElementPainters()
        {
            base.CreateElementPainters();
            fWindowPainter = new MyWindowSkinPainter(this);
        }
        protected override void OnStyleChanged(object sender, EventArgs ea) {
            base.OnStyleChanged(sender, ea);
            fWindowPainter = new MyWindowSkinPainter(this);
        }
        public override void DrawWindowCaption(DrawWindowCaptionArgs e)
        {
            MyWindowSkinPainter painter = WindowPainter as MyWindowSkinPainter;
            if (painter != null)
                painter.CurrentDockPanel = CurrentPanel;
            base.DrawWindowCaption(e);
        }
        public override void DrawApplicationCaption(DrawApplicationCaptionArgs args) {
            MyWindowSkinPainter painter = WindowPainter as MyWindowSkinPainter;
            if(painter != null)
                painter.CurrentDockPanel = CurrentPanel;
            base.DrawApplicationCaption(args);
        }

        private DockPanel _CurrentPanel;
        public DockPanel CurrentPanel
        {
            get { return _CurrentPanel; }
            set { _CurrentPanel = value; }
        }
    }
}

File: VB/E2502/MyDockManager/MyDockElementsSkinPainter.vb
Imports System
Imports DevExpress.XtraBars.Docking.Paint
Imports DevExpress.XtraBars.Styles
Imports DevExpress.XtraBars.Docking

Namespace WindowsFormsApplication1
    Public Class MyDockElementsSkinPainter
        Inherits DockElementsSkinPainter

        Public Sub New(ByVal paintStyle As SkinBarManagerPaintStyle)
            MyBase.New(paintStyle)
        End Sub

        Protected Overrides Sub CreateElementPainters()
            MyBase.CreateElementPainters()
            fWindowPainter = New MyWindowSkinPainter(Me)
        End Sub

        Protected Overrides Sub OnStyleChanged(ByVal sender As Object, ByVal ea As EventArgs)
            MyBase.OnStyleChanged(sender, ea)
            fWindowPainter = New MyWindowSkinPainter(Me)
        End Sub

        Public Overrides Sub DrawWindowCaption(ByVal e As DrawWindowCaptionArgs)
            Dim painter As MyWindowSkinPainter = TryCast(WindowPainter, MyWindowSkinPainter)
            If painter IsNot Nothing Then painter.CurrentDockPanel = CurrentPanel
            MyBase.DrawWindowCaption(e)
        End Sub

        Public Overrides Sub DrawApplicationCaption(ByVal args As DrawApplicationCaptionArgs)
            Dim painter As MyWindowSkinPainter = TryCast(WindowPainter, MyWindowSkinPainter)
            If painter IsNot Nothing Then painter.CurrentDockPanel = CurrentPanel
            MyBase.DrawApplicationCaption(args)
        End Sub

        Private _CurrentPanel As DockPanel

        Public Property CurrentPanel As DockPanel
            Get
                Return _CurrentPanel
            End Get
            Set(ByVal value As DockPanel)
                _CurrentPanel = value
            End Set
        End Property
    End Class
End Namespace
